polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -667,6 +691,11 @@ impl<T: Config> Pallet<T> {
 			disputes,
 			parent_header,
 		};
+
+		// Write back to storage only keys that have been updated or deleted.
+		for (para_id, candidates) in pending_availability_overlay.into_iter() {
```

Comment:
What about using a `Drop` instance like [here](https://github.com/paritytech/polkadot-sdk/blob/ea458d0b95d819d31683a8a09ca7973ae10b49be/polkadot/runtime/parachains/src/paras/mod.rs#L2300)?

Reply:
yes, good idea, I'll have to see how it should work in the case of this generic overlay.